test/e2e/cluster.go

```diff
@@ -115,22 +243,34 @@ var _ = Describe("Cluster", func() {
 	})
 })
 
-func createStatefulSet(ctx context.Context, cli kubernetes.Interface) error {
-	oc, _ := clients.OpenshiftClusters.Get(ctx, vnetResourceGroup, clusterName)
-	installVersion, _ := version.ParseVersion(*oc.ClusterProfile.Version)
+// clusterSubnets returns a slice containing all of the cluster subnets' resource IDs
+func clusterSubnets(oc redhatopenshift.OpenShiftCluster) []string {
+	subnetMap := map[string]struct{}{}
+	subnetMap[*oc.OpenShiftClusterProperties.MasterProfile.SubnetID] = struct{}{}
+
+	// TODO: change to workerProfileStatuses when we bump the API to 20230904 stable
+	for _, p := range *oc.OpenShiftClusterProperties.WorkerProfiles {
+		subnetMap[*p.SubnetID] = struct{}{}
```

Comment:
I might be wrong, but I think it would be safer to convert to lowercase when inserting on the map, not when inserting on the slice. Because right now we are converting to lowercase when inserting the values on the slice, which could lead to duplicates in the slice itself as the map can contain {A, a} and then slice later will contain {a, a}.

Maybe there is a precondition that this can not happen but I prefer to mention it even I am wrong.
😃  

I think it would be safer to do:
```go
func clusterSubnets(oc redhatopenshift.OpenShiftCluster) []string {
	subnetMap := map[string]struct{}{}
	subnetMap[*oc.OpenShiftClusterProperties.MasterProfile.SubnetID] = struct{}{}

	// TODO: change to workerProfileStatuses when we bump the API to 20230904 stable
	for _, p := range *oc.OpenShiftClusterProperties.WorkerProfiles {
		s := strings.ToLower(*p.SubnetID) // change -> ToLower here to ensure no duplicates
		subnetMap[s] = struct{}{}
	}

	subnets := []string{}

	for subnet := range subnetMap {
		subnets = append(subnets, subnet) //change -> ToLower NOT here
	}

	return subnets
}
```

Reply:
I don't think this is the reason E2E failed, but that makes sense and I've made the change you suggested 👍🏻